tests/store/tracking/test_file_store.py

```diff
@@ -2061,3 +2066,301 @@ def test_create_experiment_appends_to_artifact_local_path_file_uri_correctly(
+    inputs2 = sorted(inputs2, key=lambda inp: (inp.dataset.name, inp.dataset.digest))
+    assert len(inputs1) == len(inputs2)
+    for idx, inp1 in enumerate(inputs1):
+        inp2 = inputs2[idx]
+        assert dict(inp1.dataset) == dict(inp2.dataset)
+        tags1 = sorted(inp1.tags, key=lambda tag: tag.key)
+        tags2 = sorted(inp2.tags, key=lambda tag: tag.key)
+        for idx, tag1 in enumerate(tags1):
+            tag2 = tags2[idx]
+            assert tag1.key == tag1.key
+            assert tag1.value == tag2.value
+
+
+def test_log_inputs_and_retrieve_runs_behaves_as_expected(store):
+    exp_id = store.create_experiment("12345dataset")
+
+    run1 = store.create_run(
+        experiment_id=exp_id,
+        user_id="user1",
+        start_time=1,
+        tags=[],
+        run_name=None,
+    )
+    run2 = store.create_run(
+        experiment_id=exp_id,
+        user_id="user2",
+        start_time=3,
+        tags=[],
+        run_name=None,
+    )
+    run3 = store.create_run(
+        experiment_id=exp_id,
+        user_id="user3",
+        start_time=2,
+        tags=[],
+        run_name=None,
+    )
+
+    dataset1 = Dataset(
+        name="name1",
+        digest="digest1",
+        source_type="st1",
+        source="source1",
+        schema="schema1",
+        profile="profile1",
+    )
+    dataset2 = Dataset(
+        name="name2",
+        digest="digest2",
+        source_type="st2",
+        source="source2",
+        schema="schema2",
+        profile="profile2",
+    )
+    dataset3 = Dataset(
+        name="name3",
+        digest="digest3",
+        source_type="st3",
+        source="source3",
+        schema="schema3",
+        profile="profile3",
+    )
+
+    tags1 = [InputTag(key="key1", value="value1"), InputTag(key="key2", value="value2")]
+    tags2 = [InputTag(key="key3", value="value3"), InputTag(key="key4", value="value4")]
+    tags3 = [InputTag(key="key5", value="value5"), InputTag(key="key6", value="value6")]
+
+    inputs_run1 = [DatasetInput(dataset1, tags1), DatasetInput(dataset2, tags1)]
+    inputs_run2 = [DatasetInput(dataset1, tags2), DatasetInput(dataset3, tags3)]
+    inputs_run3 = [DatasetInput(dataset2, tags3)]
+
+    store.log_inputs(run1.info.run_id, inputs_run1)
+    store.log_inputs(run2.info.run_id, inputs_run2)
+    store.log_inputs(run3.info.run_id, inputs_run3)
+
+    run1 = store.get_run(run1.info.run_id)
+    assert_dataset_inputs_equal(run1.inputs.dataset_inputs, inputs_run1)
+    run2 = store.get_run(run2.info.run_id)
+    assert_dataset_inputs_equal(run2.inputs.dataset_inputs, inputs_run2)
+    run3 = store.get_run(run3.info.run_id)
+    assert_dataset_inputs_equal(run3.inputs.dataset_inputs, inputs_run3)
+
+    search_results_1 = store.search_runs(
+        [exp_id], None, ViewType.ALL, max_results=4, order_by=["start_time ASC"]
+    )
+    run1 = search_results_1[0]
+    assert_dataset_inputs_equal(run1.inputs.dataset_inputs, inputs_run1)
+    run2 = search_results_1[2]
```

Comment:
how do we know that `search_results_1` will return in  this order of run1, run3, run2?

Reply:
because we're ordering by `start_time ASC`, and all the runs have different start times hardcoded above